src/tablestore.rs

```diff
@@ -83,11 +87,15 @@ impl TableStore {
         Self {
             object_store: object_store.clone(),
             sst_format,
-            root_path,
+            root_path: root_path.clone(),
             wal_path: "wal",
             compacted_path: "compacted",
             fp_registry,
             filter_cache: RwLock::new(HashMap::new()),
+            transactional_wal_store: Arc::new(DelegatingTransactionalObjectStore::new(
+                root_path.child("wal"),
```

Comment:
Seems like it might be easier to set root_path to "/" so we can use `self.path(id)` where it was deleted on line 156.